### ISSUE TITLE
[Bug]: 310P 上 Qwen3.6-27B-w8a8 无法启用压缩 Mask

### ISSUE THREAD

**Issue by pjqiuqiu** [vllm-project/vllm-ascend#11907]
### Your current environment

Hardware:
  - NPU: Ascend 310P3 (Atlas 300I), 4 cards (TP=4)
  - Host OS: Linux (Docker)

Software
  - vLLM: 0.22.1
  - PyTorch: 2.10.0
  - torch_npu: 2.10.0 built with Ascend/op-plugin master
    (exposes _npu_flash_attention_v3 / _npu_paged_attention_splitfuse_v2)
  - CANN Toolkit: 9.1.0-beta.3
  - ascend-cann-310p-ops: 9.1.0-beta.3
  - NNAL/ATB: 9.1.0-beta.3
  - Python: 3.12.x
  - Model: Qwen3.6-27B-w8a8 (Qwen3_5ForConditionalGeneration / multimodal with ViT)

Key local patch used during test:
  - vllm_ascend/utils.py::_should_trans_nz
    - is_310p() -> True  : ViT hits aclnnMatmulWeightNz (unsupported on 310P)
    - is_310p() -> False : W8A8 hits aclnnQuantMatmulV5 error 161002

Compressed-mask ops check:
  hasattr(torch_npu, "_npu_flash_attention_v3") == True
  hasattr(torch_npu, "_npu_paged_attention_splitfuse_v2") == True

### 🐛 Describe the bug

一、问题概述
在 Atlas 300I / Ascend 310P3 上部署 Qwen3.6-27B-w8a8 时，压缩 mask（compressed mask）无法与该模型在同一套软件栈上稳定同时启用。
压缩 mask 本身在 310P 上已有算子路径，但与该模型的 W8A8 量化 + 多模态 ViT 非量化层对权重 NZ 布局的需求互相冲突。

二、压缩 Mask 启用条件（310P）
vllm-ascend 通过检测以下两个算子决定是否走压缩 mask：
torch_npu._npu_flash_attention_v3
torch_npu._npu_paged_attention_splitfuse_v2
完整依赖通常为：
组件	要求
CANN Toolkit	9.1.0-beta.3
ascend-cann-310p-ops	9.1.0-beta.3
NNAL/ATB	配套 beta.3
torch_npu	需带 op-plugin master 绑定（多数官方镜像自带包没有上述两个 API）

官方常见镜像仍为 CANN 9.1.0.BETA1 时，压缩 mask 不会生效，只能走稠密 L×L mask + ND→NZ，长上下文显存暴涨（例如 64K 约 8GB mask，ND2NZ 峰值约 16GB+）。

三、为何在 Qwen3.6-27B-w8a8 上不好使
不是模型“不支持压缩 mask 语义”，而是启用压缩 mask 所需的自编译 torch_npu，与该模型推理路径不兼容。
3.1 模型结构特点
LLM 主体：Ascend ModelSlim W8A8（npu_quant_matmul / aclnnQuantMatmulV5）
视觉侧：ViT 等非量化 FP16 linear（F.linear / unquantized_gemm）
310P 上 vllm-ascend 默认 _should_trans_nz() 对 310P 强制 True（权重转 FRACTAL_NZ）
3.2 实测死结（同一套带 v3 的自编译 torch_npu）
_should_trans_nz（310P）	现象
True（默认）	ViT / 非量化层走 aclnnMatmulWeightNz → 报错：Weight NZ unsupported on Ascend310P
False（绕开 MatmulWeightNz）	W8A8 量化 GEMM → aclnnQuantMatmulV5 error 161002

因此：开 NZ 则视觉塔挂；关 NZ 则量化 LLM 挂。压缩 mask 算子可以存在（hasattr 为 True），但服务无法在该模型上完整跑通。
对比：同环境下 PaddleOCR-VL（FP16、非 W8A8）在 return False + 自编译 torch_npu 上可启动，且压缩 mask 可用——说明问题集中在 W8A8 + 多模态非量化层 + 自编译 torch_npu 的 NZ/Quant 路径，而非单纯 attention mask 逻辑。
3.3 不启用压缩 Mask 时的折中
改用官方配套栈（例如 vLLM 0.18.0 + 官方 torch_npu + 较旧 CANN，如 9.0.T3）时，Qwen3.6-27B-w8a8 可正常启动，但通常没有 _npu_flash_attention_v3，长上下文仍受稠密 mask 显存限制（维护者建议先 20K，gpu-memory-utilization 0.85）。

四、根因归纳
压缩 mask 与可用商业镜像脱节：算子在 CANN 9.1.0-beta.3 + op-plugin 新绑定中，BETA1/官方 torch_npu 常缺 v3/v2。
为启用压缩 mask 自编译的 torch_npu 在 310P 上 NZ matmul 不完整：MatmulWeightNz 对当前 SOC 不支持/不可用。
W8A8 量化 matmul 又依赖 NZ 权重布局：关掉 NZ 后 QuantMatmulV5 返回 161002。
Qwen3.6-27B-w8a8 = 量化 LLM + 非量化 ViT，对 NZ 开关无法统一，形成互斥。
期望官方提供同一套 310P 发行版，同时满足：压缩 mask（FA v3 / PA splitfuse v2）、310P 可用的 QuantMatmul（W8A8）、多模态非量化 linear（不依赖不可用的 MatmulWeightNz，或提供 310P 可用替代）。

五、诉求
请官方确认并给出官方支持组合：
310P3 + Qwen3.6-27B-w8a8 推荐的 CANN / 310p-ops / NNAL / torch_npu / vllm-ascend 版本矩阵
该矩阵下是否支持压缩 mask；若不支持，官方建议的 max-model-len 上限
是否计划修复：自编译/新版 torch_npu 在 310P 上 MatmulWeightNz vs QuantMatmulV5 的互斥问题，使 W8A8 多模态模型也可启用压缩 mask

六、一句话总结
310P 压缩 mask 依赖新算子与自编译 torch_npu；该 torch_npu 在 310P 上要么 ViT 踩 MatmulWeightNz 不支持，要么关 NZ 后 W8A8 踩 QuantMatmulV5 161002。Qwen3.6-27B-w8a8 同时含非量化视觉层与量化 LLM，无法在同一套栈上同时开压缩 mask 并稳定推理；官方 BETA1 镜像本身也还不支持压缩 mask。

**Comment by ZT-AIA** [comment]
@Tflowers-0129  Plz see see this issue.

**Comment by Tflowers-0129** [comment]
> 若不支持，官方建议的 max-model-len 上限
> 是否计划修复

算子不支持，在我们看来这就不是一个切实存在的问题；而是一个待优化的特性，目前外部可用的torch-npu和CANN版本还没有压缩mask特性，需要等待CANN和torch-npu商业版本发布

**Comment by pjqiuqiu** [comment]
> > 若不支持，官方建议的 max-model-len 上限
> > 是否计划修复
> 
> 算子不支持，在我们看来这就不是一个切实存在的问题；而是一个待优化的特性，目前外部可用的torch-npu和CANN版本还没有压缩mask特性，需要等待CANN和torch-npu商业版本发布

大佬看一下这个，https://huggingface.co/adeepv/Qwen3.6-27B-W8A16-Ascend310P
可以开到256k，原理是按块改 mask 大小，而不是 O(LxL)
